packages/near-lake-primitives/README.md

```diff
@@ -0,0 +1,345 @@
+
+### Important Notes on `Block`
+
+- All the entities located on different shards were merged into one single list without differentiation.
+- `Block` is not the fairest name for this structure either. NEAR Protocol is a sharded blockchain, so its block is actually an ephemeral structure that represents a collection of real blocks called chunks in NEAR Protocol.
+
+### `BlockHeader`
+
+Replacement for `BlockHeaderView` from `near-primitives`. Shrunken and simplified. Note: the original `BlockHeaderView` is still accessible via the `.streamerMessage` attribute.
+
+#### `BlockHeader` Structure Definition
+
+```ts
+export class BlockHeader {
+    constructor(
+        readonly height: number,
+        readonly hash: string,
+        readonly prevHash: string,
+        readonly author: string,
+        readonly timestampNanosec: string,
+        readonly epochId: string,
+        readonly nextEpochId: string,
+        readonly gasPrice: string,
+        readonly totalSupply: string,
+        readonly latestProtocolVersion: number,
+        readonly randomValue: string,
+        readonly chunksIncluded: number,
+        readonly validatorProposals: ValidatorStakeView[]) { 
+        }
+    ... // helper method ommitted for brevity
+}
+```
+
+### `Receipt`
+
+This field is a simplified representation of the `ReceiptView` structure from `near-primitives`.
+
+#### `Receipt` Structure Definition
+
+```ts
+export class Receipt implements Events {
+  constructor(
+    readonly receiptKind: ReceiptKind, 
+    readonly receiptId: string, 
+    readonly receiverId: string, 
+    readonly predecessorId: string, 
+    readonly status: ExecutionStatus, 
+    readonly executionOutcomeId?: string | undefined, 
+    readonly logs: string[] = []) {
+    }
+  ... // helper methods ommitted for brevity
+}
+```
+
+#### `Receipt` Fields
+
+##### `receiptKind`
+
+Defined the type of the `Receipt`: `Action` or `Data` representing the `ActionReceipt` and `DataReceipt`.
+
+##### `receiptId`
+
+The ID of the `Receipt` of the `CryptoHash` type.
+
+##### `receiverId`
+
+The receiver account id of the `Receipt`.
+
+##### `predecessorId`
+
+The predecessor account id of the `Receipt`.
+
+##### `status`
+
+Represents the status of `ExecutionOutcome` of the `Receipt`.
+
+See the `ExecutionStatus` enum section for the details.
+
+##### `executionOutcomeId`
+
+The id of the `ExecutionOutcome` for the `Receipt`. Returns `null` if the `Receipt` isn’t executed yet and has a postponed status.
+
+##### `logs`
+
+The original logs of the corresponding `ExecutionOutcome` of the `Receipt`.
+
+Note: not all of the logs might be parsed as JSON Events (`Events`).
+
+#### `Receipt` Helper Methods
+
+```ts
+export class Receipt {
+    ... // constructor ommitted for brevity
+    get events(): Event[] {}
+
+    static fromOutcomeWithReceipt(outcomeWithReceipt: OutcomeWithReceipt): Receipt {}
+}
+```
+
+##### `Receipt.events()`
```

Comment:
Is it possible to provide the expected return type as you did for the `Block` methods?

Reply:
I'm confused. I see this explained below:

`Returns an Array of `Events` for the `Receipt`, if any. This might be empty if the `logs` field is empty or doesn’t contain JSON Events compatible log records.`

Reply:
Sorry I mean adding it to the method title 

Like this `.events(): Events[]`